react/features/stream-effects/blur/index.js

```diff
@@ -27,11 +27,17 @@ export function createBlurEffect() {
         return Promise.reject(new Error('JitsiStreamBlurEffect not supported!'));
     }
 
-    tfc.setBackend(webGlBackend, true);
     if (tfc.getBackend() !== webGlBackend) {
-        console.debug('TensorFlow backend could not be changed to use WebGL');
+        tfc.setBackend(webGlBackend).then(() => {
+            console.info('TensorFlow backend changed to use WebGL');
 
-        return Promise.reject(new Error('JitsiStreamBlurEffect not supported!'));
+            return bpModelPromise.then(bpmodel => new JitsiStreamBlurEffect(bpmodel));
+        })
+        .catch(err => {
+            console.error('TensorFlow backend could not be changed to use WebGL: ', err);
+
+            return Promise.reject(new Error('JitsiStreamBlurEffect not supported!'));
+        });
```

Comment:
I don't think this will return the rejected promise. What will get returned is the promise below. Maybe what you want to do is something like this:
```
const setBackendPromise = new Promise((resolve, reject) => {
  if (webGLBackend) {
    resolve();
    return;
  }

  return tfc.setBackend(webGlBackend)
    .then(resolve, reject)
})
return setBackendPromise
  .then(() => bpModelPromise)
  .then(bpmodel => new JitsiStreamBlurEffect(bpmodel))
```